crates/driver/src/tests/setup/blockchain.rs

```diff
@@ -684,46 +860,68 @@ impl Blockchain {
+                    execution: execution.clone(),
+                    interactions: (!is_jit_order)
+                        .then(|| {
+                            vec![
+                                Interaction {
+                                    address: sell_token.address(),
+                                    calldata: match solution.calldata {
+                                        super::Calldata::Valid { additional_bytes } => {
+                                            transfer_interaction
+                                                .into_iter()
+                                                .chain(
+                                                    std::iter::repeat(0xab).take(additional_bytes),
+                                                )
+                                                .collect()
+                                        }
+                                        super::Calldata::Invalid => vec![1, 2, 3, 4, 5],
+                                    },
+                                    inputs: Default::default(),
+                                    outputs: Default::default(),
+                                    internalize: false,
+                                },
+                                Interaction {
+                                    address: pair.contract.address(),
+                                    calldata: match solution.calldata {
+                                        super::Calldata::Valid { .. } => swap_interaction,
+                                        super::Calldata::Invalid => {
+                                            vec![10, 11, 12, 13, 14, 15, 63, 78]
+                                        }
+                                    },
+                                    inputs: vec![eth::Asset {
+                                        token: sell_token.address().into(),
+                                        // Surplus fees stay in the contract.
+                                        amount: (execution.sell - order.surplus_fee()).into(),
+                                    }],
+                                    outputs: vec![eth::Asset {
+                                        token: buy_token.address().into(),
+                                        amount: execution.buy.into(),
+                                    }],
+                                    internalize: order.internalize,
+                                },
+                            ]
+                        })
+                        .unwrap_or_default(),
+                },
+                is_jit_order,
+            ));
+        }
+        Solution {
+            trades: fulfillments
+                .into_iter()
+                .map(|(order, is_jit_order)| {
+                    if *is_jit_order {
+                        Trade::Jit(order)
+                    } else {
+                        Trade::Fulfillment(order)
+                    }
```

Comment:
We have access to the `is_jit_order` flag when the Fulfillment is being built. This logic can be removed, then. Also, it is very hard to read how Fulfillment interactions are constructed; it may make sense to extract the logic into a separate variable.